azext_iot/tests/iothub/state/conftest.py

```diff
@@ -0,0 +1,523 @@
+    user_id = list(provisioned_iothubs[0]["hub"]["identity"]["userAssignedIdentities"].keys())[0]
+    sub_id = provisioned_iothubs[0]["hub"]["subscriptionid"]
+
+    # mapping of scope to role
+    scope_dict = {
+        provisioned_storage["storage"]["id"]: "Storage Blob Data Contributor",
+        provisioned_event_hub["eventhub"]["id"]: "Azure Event Hubs Data Sender",
+        provisioned_service_bus["queue"]["id"]: "Azure Service Bus Data Sender",
+        provisioned_service_bus["topic"]["id"]: "Azure Service Bus Data Sender"
+    }
+    for scope, role in scope_dict.items():
+        _assign_rbac_role(assignee=user_principal_id, scope=scope, role=role)
+        if use_system_endpoints:
+            for hub_principal_id in hub_principal_ids:
+                _assign_rbac_role(assignee=hub_principal_id, scope=scope, role=role)
+    sleep(30)
+
+    suffix = "systemid" if use_system_endpoints else "userid"
+    user_identity_parameter = "" if use_system_endpoints else f"--identity {user_id}"
+    cli.invoke(
+        f"iot hub routing-endpoint create -n eventhub-{suffix} -r {hub_rg} -g {hub_rg} -s {sub_id}"
+        f" -t {EndpointType.EventHub.value} --hub-name {hub_name} --endpoint-uri {eventhub_endpoint_uri}"
+        f" --entity-path {eventhub_name} --auth-type identityBased {user_identity_parameter}"
+    )
+
+    cli.invoke(
+        f"iot hub routing-endpoint create -n queue-{suffix} -r {hub_rg} -g {hub_rg} -s {sub_id}"
+        f" -t {EndpointType.ServiceBusQueue.value} --hub-name {hub_name} --endpoint-uri "
+        f"{servicebus_endpoint_uri} --entity-path {servicebus_queue} --auth-type identityBased "
+        f"{user_identity_parameter}"
+    )
+
+    cli.invoke(
+        f"iot hub routing-endpoint create -n topic-userid -r {hub_rg} -g {hub_rg} -t {EndpointType.ServiceBusTopic.value} "
+        f"--hub-name {hub_name} -s {sub_id} --endpoint-uri {servicebus_endpoint_uri} --entity-path {servicebus_topic} "
+        f"--auth-type identityBased --identity {user_id}"
+    )
+
+    cli.invoke(
+        f"iot hub routing-endpoint create -n storagecontainer-key -r {hub_rg} -g {hub_rg} -t "
+        f"{EndpointType.AzureStorageContainer.value} --hub-name {hub_name} -c {storage_cstring} -s {sub_id} "
+        f"--container {storage_container}  -b 350 -w 250 --encoding json")
+
+    # add routes - prob change one to be custom endpoint
+    cli.invoke(
+        f"iot hub route create --endpoint events --hub-name {hub_name} -g {hub_rg}"
+        f" --name {generate_generic_id()} --source devicelifecycleevents --condition false --enabled true"
+    )
+    cli.invoke(
+        f"iot hub route create --endpoint events --hub-name {hub_name} -g {hub_rg}"
+        f" --name {generate_generic_id()} --source twinchangeevents --condition true --enabled false"
+    )
+
+    # add certificate
+    cert = create_self_signed_certificate(subject="aziotcli", valid_days=1, cert_output_dir=None)["certificate"]
+    print(cert)
```

Comment:
do we need this print, or was it just for debugging purposes?

Reply:
debugging lol